src/models/state/mempool.rs

```diff
@@ -59,7 +60,24 @@ pub const TRANSACTION_NOTIFICATION_AGE_LIMIT_IN_SECS: u64 = 60 * 60 * 24;
 
 type LookupItem<'a> = (Digest, &'a Transaction);
 
-#[derive(Debug, Clone, PartialEq, Eq, GetSize)]
+/// Represents a mempool state change.
+///
+/// For purpose of notifying interested parties
+#[derive(Debug, Clone)]
+pub enum MempoolEvent {
+    /// a transaction was added to the mempool
+    AddTx(Transaction),
+
+    /// a transaction was removed from the mempool
+    RemoveTx(Transaction),
+
+    /// the mutator-set of a transaction was updated in the mempool.
+    ///
+    /// (Digest of Tx before update, Tx after mutator-set updated)
+    UpdateTxMutatorSet(Digest, Transaction),
+}
+
+#[derive(Debug, GetSize)]
```

Comment:
Two comments:
1. Our files are growing pretty big. Can we have a separate file for this datatype?
2. The MempoolEvent should only contain a `TransactionKernel`. It doesn't need the transaction proof for anything, as the mempool already has that (where it's used for sharing transactions with peers and merging transactions when mining). The `MempoolEvent`s received by the wallet only need the `TransactionKernel`.